qiskit/compiler/transpiler.py

```diff
@@ -5,46 +5,322 @@
+                    [qr[0], None, None, qr[1], None, qr[2]]
+
+        seed_transpiler (int):
+            sets random seed for the stochastic parts of the transpiler
+
+        optimization_level (int):
+            How much optimization to perform on the circuits.
+            Higher levels generate more optimized circuits,
+            at the expense of longer transpilation time.
+                0: no optimization
+                1: light optimization
+                2: heavy optimization
+
+        transpile_config (TranspileConfig):
+            transpiler configuration, containing some or all of the above options.
+            If any other options is explicitly set (e.g. coupling_map), it
+            will override the transpile_config's.
+
+        backend (BaseBackend):
+            If set, transpiler options are automatically grabbed from
+            backend.configuration() and backend.properties().
+            If any other option is explicitly set (e.g. coupling_map), it
+            will override the backend's.
+            If any other options is set in the transpile_config, it will
+            also override the backend's.
+            Note: the backend arg is purely for convenience. The resulting
+                circuit may be run on any backend as long as it is compatible.
+
+        seed_mapper (int):
+            DEPRECATED in 0.8: use ``seed_transpiler`` kwarg instead
+
+        pass_manager (PassManager):
+            DEPRECATED in 0.8: use ``pass_manager.run(circuits)`` directly
+
+    Returns:
+        QuantumCircuit or list[QuantumCircuit]: transpiled circuit(s).
+
+    Raises:
+        TranspilerError: in case of bad inputs to transpiler or errors in passes
+    """
+    # Deprecation matter
+    if seed_mapper:
+        warnings.warn("seed_mapper has been deprecated and will be removed in the "
+                      "0.9 release. Instead use seed_transpiler to set the seed "
+                      "for all stochastic parts of the.", DeprecationWarning)
+        seed_transpiler = seed_mapper
+
+    if pass_manager:
+        warnings.warn("After the 0.9 release, the transpile() function no longer accepts "
+                      "a pass_manager arg. Instead use pass_manager.run(circuits) directly "
+                      "to run a custom pipeline of passes.", DeprecationWarning)
+        return pass_manager.run(circuits)
+
+    # Get TranspileConfig(s) to configure the circuit transpilation job(s)
+    circuits = circuits if isinstance(circuits, list) else [circuits]
+    transpile_configs = _parse_transpile_args(circuits, basis_gates, coupling_map,
+                                              backend_properties, initial_layout,
+                                              seed_transpiler, optimization_level,
+                                              transpile_config, backend)
+
+    # Transpile circuits in parallel
+    circuits = parallel_map(_transpile_circuit, list(zip(circuits, transpile_configs)))
```

Comment:
If I understand correctly, before this the default pass manager should be selected and then it should be run. The parallelization should happen inside the `run` method?

Reply:
The `_transpile_circuit` function chooses the pass manager (based on the transpile_config).

Since I am allowing each circuit to have a different transpile_config, this is the correct place to parallelize, because there may be separate pass managers associated with each circuit.

The issue about parallelizing `PassManager.run()` (#2156) is a separate use case, where the user explicitly says that they want to tranpile a batch of circuits using the same pass manager.